tests/equation_bcs/test_equation_bcs_assemble.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+
+from firedrake import *
+
+
+def test_equation_bcs_direct_assemble_one_form():
+    mesh = UnitSquareMesh(2, 2)
+    V = FunctionSpace(mesh, "CG", 1)
+    u = Function(V)
+    v = TestFunction(V)
+    F = - inner(grad(u), grad(v)) * dx
+    F1 = inner(u, v) * ds(1)
+    bc = EquationBC(F1 == 0, u, 1)
+
+    assemble(F, bcs=extract_equation_bc_forms(bc, 'F'))
+    assemble(F, bcs=bc)
```

Comment:
Can we check for something more than "the code runs"?

Reply:
Now checks assembled values.